dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/loadbalance/LeastActiveLoadBalance.java

```diff
@@ -103,8 +99,7 @@
             // Return a invoker based on the random value.
             for (int i = 0; i < leastCount; i++) {
                 int leastIndex = leastIndexes[i];
-                offsetWeight -= weights[leastIndex];
-                if (offsetWeight < 0) {
+                if (offsetWeight < weights[leastIndex]) {

```

Comment:
It will cause problem, The logic isn't same with RandomLoadBalance. Here, the weights just record self weight.